### PR TITLE
request tracing & notification services initial commit

### PR DESCRIPTION
**Pull request by mulderbaba**

### CODE REVIEW

**Comment by smillidge** [comment]
jenkins test please

File: nucleus/payara-modules/requesttracing-core/src/main/java/fish/payara/nucleus/requesttracing/RequestEventStore.java [hunk 1]
```diff
@@ -0,0 +1,66 @@
...
+    };
+
+    public void storeEvent(RequestEvent requestEvent) {
+        if (!EventType.TRACE_START.equals(requestEvent.getEventType())) {
+            RequestEvent startEvent = eventStore.get().get(EventType.TRACE_START);
+            if (startEvent != null) {
+                requestEvent.setConversationId(startEvent.getId());
+            }
+        }
+        eventStore.get().put(requestEvent.getEventType(), requestEvent);
```
Comment (smillidge) [inline nucleus/payara-modules/requesttracing-core/src/main/java/fish/payara/nucleus/requesttracing/RequestEventStore.java]: How do you maintain order in the Events with respect to timestamp?

Comment (mulderbaba) [inline nucleus/payara-modules/requesttracing-core/src/main/java/fish/payara/nucleus/requesttracing/RequestEventStore.java]: RequestEvent implements Comparable and TreeMap maintains the order according to that.

Comment (smillidge) [inline nucleus/payara-modules/requesttracing-core/src/main/java/fish/payara/nucleus/requesttracing/RequestEventStore.java]: OK cool

**Comment by danPowell52** [comment]
jenkins test please

**Comment by Pandrex247** [comment]
Has some conflicts with the new phonehome service in various poms - doesn't look like there's any *actual* conflicts, it's just git being fussy about changes being too close to each other.

**Comment by mulderbaba** [comment]
It depicts a circular reference since I added a dependency to kernel, trying to resolve it now.

**Comment by danPowell52** [comment]
jenkins test please

**Comment by Pandrex247** [comment]
jenkins test please

**Comment by Pandrex247** [comment]
jenkins test please

**Comment by Pandrex247** [comment]
jenkins test please

**Comment by danPowell52** [comment]
jenkins test please

**Comment by mulderbaba** [comment]
jenkins test please

**Comment by smillidge** [comment]
jenkins test please

**Comment by mulderbaba** [comment]
jenkins test please

**Comment by mulderbaba** [comment]
jenkins test please

**Comment by mulderbaba** [comment]
jenkins test please

**Comment by Pandrex247** [comment]
jenkins test please

**Comment by mulderbaba** [comment]
jenkins test please

**Comment by mulderbaba** [comment]
jenkins test please

**Pull request merged**